tests/t0123_gateway_json_cbor_test.go

```diff
@@ -1,13 +1,858 @@
+				Request: Request().
+					Path("ipfs/%s", dagTraversalCID).
+					Query("filename", fmt.Sprintf("foobar.%s", row.Format)).
+					Header("Accept", fmt.Sprintf("application/vnd.ipld.dag-%s", row.Format)),
+				Response: Expect().
+					Headers(
+						Header("Content-Disposition").
+							Hint("includes Content-Disposition").
+							Contains("%s; filename=\"foobar.%s\"", row.Disposition, row.Format),
+					),
+			},
+			/**
+			  test_expect_success "GET for application/vnd.ipld.dag-$format with ?download=true forces Content-Disposition: attachment" '
+			  curl -svX GET -H "Accept: application/vnd.ipld.dag-$format" "http://127.0.0.1:$GWAY_PORT/ipfs/$CID?filename=foobar.$format&download=true" >/dev/null 2>curl_output_filename &&
+			  test_should_contain "< Content-Disposition: attachment" curl_output_filename
+			  '
+			*/
+			{
+				Name: fmt.Sprintf("GET for application/vnd.ipld.dag-%s with ?download=true forces Content-Disposition: attachment", row.Format),
+				Request: Request().
+					Path("ipfs/%s", dagTraversalCID).
+					Query("filename", fmt.Sprintf("foobar.%s", row.Format)).
+					Query("download", "true").
+					Header("Accept", fmt.Sprintf("application/vnd.ipld.dag-%s", row.Format)),
+				Response: Expect().
+					Headers(
+						Header("Content-Disposition").
+							Hint("includes Content-Disposition").
+							Contains("attachment; filename=\"foobar.%s\"", row.Format),
+					),
+			},
+			/**
+			  # Cache control HTTP headers
+			  # (basic checks, detailed behavior is tested in  t0116-gateway-cache.sh)
+
+			  test_expect_success "GET response for application/vnd.ipld.dag-$format includes Etag" '
+			  test_should_contain "< Etag: \"${CID}.dag-$format\"" curl_output
+			  '
+			  test_expect_success "GET response for application/vnd.ipld.dag-$format includes X-Ipfs-Path and X-Ipfs-Roots" '
+			  test_should_contain "< X-Ipfs-Path" curl_output &&
+			  test_should_contain "< X-Ipfs-Roots" curl_output
+			  '
+			  test_expect_success "GET response for application/vnd.ipld.dag-$format includes Cache-Control" '
+			  test_should_contain "< Cache-Control: public, max-age=29030400, immutable" curl_output
+			  '
+			*/
+			{
+				Name: fmt.Sprintf("Cache control HTTP headers (%s)", row.Format),
+				Hint: `(basic checks, detailed behavior is tested in t0116-gateway-cache.sh)`,
+				Request: Request().
+					Path("ipfs/%s", dagTraversalCID).
+					Header("Accept", fmt.Sprintf("application/vnd.ipld.dag-%s", row.Format)),
+				Response: Expect().
+					Headers(
+						Header("Etag").Hint("includes Etag").Contains("%s.dag-%s", dagTraversalCID, row.Format),
+						Header("X-Ipfs-Path").Hint("includes X-Ipfs-Path").Exists(),
+						Header("X-Ipfs-Roots").Hint("includes X-Ipfs-Roots").Exists(),
+						Header("Cache-Control").Hint("includes Cache-Control").Contains("public, max-age=29030400, immutable"),
```

Comment:
TODO for later, we're going to want to go through all those exact cache-control checks and make them more lenient most likely.

Reply:
👍 we have https://github.com/ipfs/gateway-conformance/issues/21 to capture this.